### PR TITLE
chore(website): configure eslint with docusaurus, react-ts and prettier plugins

### LINKED ISSUE #80: Configure eslint for website
Docusaurus provides an eslint plugin: https://docusaurus.io/docs/api/misc/@docusaurus/eslint-plugin

Also this article describes how to implement eslint for a react / typescript application: https://dev.to/anukr98/setting-up-eslint-with-tsjs-in-your-react-project-in-2023-57o

### PR DESCRIPTION
**Pull request by MridulDhiman**
Closes #80 

## 📑 Description
This PR configures ESLint with Docusaurus, React-TypeScript, and Prettier plugins. It ensures that Prettier plugins do not conflict with the root directory.

## ✅ Checks
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required

## ℹ Additional Information
The following changes have been made:

1. Added ESLint configuration (`eslint.config.mjs`)
2. Created `.eslintignore` and `.prettierignore` files
3. Added `.prettierrc.mjs` for Prettier configuration in the website folder
4. Updated `package.json` with new scripts:
   * lint: `eslint .`
   * lint:fix: `eslint --fix .`
   * format: `prettier --write .`
5. Installed the following dev dependencies:
   * `@eslint/js@^9.5.0`
   * `eslint@^8.57.0`
   * `eslint-plugin-import@^2.29.1`
   * `eslint-plugin-prettier@^5.1.3`
   * `eslint-plugin-react@^7.34.3`
   * `globals@^15.6.0`
   * `prettier@^3.3.2`
   * `@docusaurus/eslint-plugin@^3.4.0`
   * `typescript-eslint@^7.14.1`

### CODE REVIEW

File: website/eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import pluginJs from "@eslint/js";
+import * as tseslint from "typescript-eslint";
+import pluginReactConfig from "eslint-plugin-react/configs/recommended.js";
+import prettierEslintPlugin from "eslint-plugin-prettier";
+import docusaurusEslintPlugin from "@docusaurus/eslint-plugin";
+
+export default [
+  // TODO: eslint configure for .md, .mdx files
+  {files: ["*/.{js,mjs,cjs,ts,jsx,tsx}"]},
+  {ignores: ["babel.config.js", "eslint.config.mjs", "docusaurus.config.ts"]},
```
[the remark this answers is not in the document]
Comment (MridulDhiman) [reply website/eslint.config.mjs]: Hey @kosmoz, you can check latest changes.

File: website/eslint.config.mjs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  {
+    settings: {
+      react: {
+        version: "detect",
+      },
+    },
+  },
+  {
+    rules: {
+      "prettier/prettier": "off",
```
Comment (kosmoz) [inline website/eslint.config.mjs:43]: If I understand correctly, this turns off prettier eslint rules. If that is the case, I'm not sure why the prettier plugin is needed at all.

File: website/eslint.config.mjs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import pluginJs from "@eslint/js";
+import * as tseslint from "typescript-eslint";
+import pluginReactConfig from "eslint-plugin-react/configs/recommended.js";
+import prettierEslintPlugin from "eslint-plugin-prettier";
+import docusaurusEslintPlugin from "@docusaurus/eslint-plugin";
+
+export default [
+  // TODO: eslint configure for .md, .mdx files
+  {files: ["*/.{js,mjs,cjs,ts,jsx,tsx}"]},
+  {ignores: ["babel.config.js", "eslint.config.mjs", "docusaurus.config.ts", ".docusaurus/*", "build/*"]},
```
Comment (kosmoz) [inline website/eslint.config.mjs:11]: I actually think the config files should be included

File: website/eslint.config.mjs [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import globals from "globals";
+import pluginJs from "@eslint/js";
+import * as tseslint from "typescript-eslint";
+import pluginReactConfig from "eslint-plugin-react/configs/recommended.js";
+import prettierEslintPlugin from "eslint-plugin-prettier";
+import docusaurusEslintPlugin from "@docusaurus/eslint-plugin";
+
+export default [
+  // TODO: eslint configure for .md, .mdx files
+  {files: ["*/.{js,mjs,cjs,ts,jsx,tsx}"]},
```
Comment (kosmoz) [inline website/eslint.config.mjs:10]: this doesn't seem to do anything, I think it can be removed

File: website/eslint.config.mjs [hunk 5]
```diff
@@ -0,0 +1,50 @@
+import globals from "globals";
+import pluginJs from "@eslint/js";
+import * as tseslint from "typescript-eslint";
+import pluginReactConfig from "eslint-plugin-react/configs/recommended.js";
+import prettierEslintPlugin from "eslint-plugin-prettier";
+import docusaurusEslintPlugin from "@docusaurus/eslint-plugin";
+
+export default [
```
Comment (kosmoz) [inline website/eslint.config.mjs:8]: please add eslint type definitions: `npm i --save-dev @types/eslint`

```suggestion
/** @type {import('eslint').Linter.FlatConfig[]} */
export default [
```

File: website/eslint.config.mjs (hunk 2 above)
Comment (MridulDhiman) [reply website/eslint.config.mjs]: Actually, I was testing for unusual logs while running `npm run lint`, and it was giving some 1000+ logs, so I had disabled it temporarily for debugging purposes. Forgot to change it to `warn`. Updated in the latest commit.

**Review by kosmoz — APPROVED** [review]

**Pull request merged**